src/isar/state_machine/state_machine.py

```diff
@@ -298,28 +301,19 @@ def _log_state_transition(self, next_state):
 
     def log_step_overview(self, mission: Mission):
         """Log an overview of the steps in a mission"""
-        step_status: str = "\n".join(
-            [
-                f"{i:>3}  {type(step).__name__:<20} "
-                f"{str(step.id)[:8]:<32} -- {step.status}"
-                for i, step in enumerate(mission.steps)
-            ]
-        )
-        self.logger.info(f"Mission step overview:\n{step_status}")
-
-    def _check_dependencies(self):
-        """Check dependencies of previous steps"""
-        if self.current_step and self.current_step.depends_on:
-            dependency_steps = [
-                step
-                for step in self.current_mission.steps
-                if step.id in self.current_step.depends_on
-            ]
-            if not all(
-                [step.status == StepStatus.Completed for step in dependency_steps]
-            ):
-                return False
-        return True
+        log_statements: List[str] = []
+        for task in mission.tasks:
+            log_statements.append(
+                f"{type(task).__name__:<20} {str(task.id)[:8]:<32} -- {task.status}"
+            )
+            for j, step in enumerate(task.steps):
+                log_statements.append(
+                    f"{j:>3} {type(step).__name__:<20} {str(step.id)[:8]:<32} -- {step.status}"
```

Comment:
Too long line, but it may make sense to keep it thus, due to its complexity. Add the `noqa` tag to ignore warnings from `flake8`
```suggestion
                    f"{j:>3} {type(step).__name__:<20} {str(step.id)[:8]:<32} -- {step.status}"  # noqa: E501
```